WalletWasabi/Wallets/FileSystemBlockRepository.cs

```diff
@@ -15,17 +16,124 @@ namespace WalletWasabi.Wallets
 	/// </summary>
 	public class FileSystemBlockRepository : IRepository<uint256, Block>
 	{
-		public FileSystemBlockRepository(string blocksFolderPath, Network network)
+		public FileSystemBlockRepository(string blocksFolderPath, Network network, long targetBlocksFolderSizeMb = 300)
 		{
 			BlocksFolderPath = blocksFolderPath;
 			Network = network;
 			CreateFolders();
+			EnsureBackwardsCompatibility();
+			Prune(targetBlocksFolderSizeMb);
```

Comment:
I would remove this prune functionality at all. The size limit is just an estimation and LastAccessTimeUtc worked bugous on cross-platform AFAIK remember. This can go wrong in many ways and the benefit is minor - no issues reported about disk space usage.

Reply:
This is important. Normally the user would never even need to fetch a block twice. The cross platform bug was that without setting it it only worked on Windows for the keymanagers. But as you know our wallet list is correct cross platform so we know for sure this works.

Reply:
For me, the wallet list works chaotically in Wallet Explorer - I just don't care about it as it will be replaced. Anything on the OS can read the file, like AV and that will mess up our internal logic.

Reply:
Sorry, I don't understand what you're saying.

Reply:
I don't understand David's last comment either but I agree to not add this prune functionality at all.
If someone has a lot of blocks then the wallet will prune the block each time Wasabi is run, and then it will download blocks when you load the wallet.